core/metacling/src/TCling.cxx

```diff
@@ -7177,7 +6906,8 @@ static std::string GetSharedLibImmediateDepsSlow(std::string lib,
          if (skipLoadedLibs && llvm::sys::DynamicLibrary::SearchForAddressOfSymbol(SymName))
             continue;
 
-         std::string found = ResolveSymbol(SymName, interp, /*searchSystem*/false);
+         R__LOCKGUARD(gInterpreterMutex);
```

Comment:
I don't understand. The lock shows as "new", introduced by this PR - so I don't see legacy. If we need it (do we?) then why don't we need it for other interpreter calls in front of the lock? We might not need it: if this is a non-public function and all callers have acquired a lock.

I trust you will revisit and decide, if needed in a subsequent PR.

Reply:
Removal is easy. Let's summon @pcanal before we do that...

Reply:
> LazyFunctionCreatorAutoloadForModule locked?

Why that function in particular?

The lock on line 6902 seems to be in the function GetSharedLibImmediateDepsSlow which is used from GetSharedLibDeps which is part of the TInterpreter interface ... hence it can be called from anywhere (no possible assumption on already held locks).

Reply:
So we keep the lock?

Reply:
Yes.